packages/axios-extension/src/auth/connection.ts

```diff
@@ -144,25 +143,33 @@ export function attachConnectionHandler(provider: ServiceProvider) {
     });
 
     // throw error if the user is unauthorized otherwise, remove interceptor if successfully connected
-    const oneTimeRespInterceptorId = provider.interceptors.response.use((response: AxiosResponse) => {
-        if (response.status === 401) {
-            throw new ConnectionError(response.statusText, response);
-        }
```

Comment:
AFAIK, client code (for example, Fiori tools app generator), relies on catching this error to show a login prompt. I know Fiori tools has not completely moved to using this library. Does it make sense to keep this behavior?

Reply:
It is still thrown. Before my change `4xx` was handled as success, and the code above was throwing it. Now, `4xx` is thrown by `axios` without us having to check and throw, so the behavior hasn't changed here for the consumer